documentation/src/docs/security/generating-tls-artifacts.md

```diff
@@ -0,0 +1,324 @@
+             -import -file ca-cert.crt \
+             -storepass changeit
+
+   # Optionally, list the truststore's contents and inspect the output to verify everything is in order. The output 
+   # should show a single entry with alias name `caroot` and entry type `trustedCertEntry`.
+   $ keytool -list -v -keystore client.truststore.jks -storepass changeit
+   ```
+
+At this point, the following CA and truststore artifacts should be ready:
+
+| File | Description | Example command for inspecting the file's Contents |
+|:-----:|:--------|:--------------|
+| `ca-cert.crt` | PEM-encoded X.509 certificate of the CA | `$ openssl x509 -in ca-cert.crt -text -noout` |
+| `ca-key.key` | PEM-encoded file containing the CA's encrypted private key  | `$ openssl pkcs8 -inform PEM -in ca-key.key -topk8` |
+| `client.truststore.jks` | A password-protected truststore file containing the CA's certificate | `$ keytool -list -v -keystore client.truststore.jks -storepass changeit` |
+
+## Generate TLS Certificates and Other TLS Artifacts for the Server
+
+Here, we prepare a set of TLS artifacts that can be used by one or more nodes.  
+
+We'll use the CA created in the previous stage to sign the server certificate for demonstration, but in real usage you 
+may instead get it signed by an existing internal CA in your environment or a public CA.
+
+The high-level steps involved in this stage are:
+
+1. Generate a set of server certificate and private key.
+2. Generate a Certificate Signing Request (CSR) for the server certificate.
+3. Submit the CSR to a CA and obtain a signed certificate.
+4. Prepare a keystore containing the signed server certificate and the CA's certificate.
+5. Export the server certificate's private key.
+
+Note that these steps will need to be repeated with different input for each set. We discussed the number of sets 
+required earlier in the [Overview](#overview) section. 
+
+The above steps are described below: 
+
+1. Generate a set of server certificate and private key.
+
+   The following command will generate a Java JKS keystore containing the following artifacts for the server:
+   * An X.509 certificate containing the public key identifying the server
+   * A private key for the public/private key pair
+
+   ```bash
+   $ keytool -storetype JKS -keystore <java-keystore-file-name> -storepass <java-keystore-password>\
+           -genkey -keyalg <key-algorithm> -keysize <key-size> -keypass <private-key-password>\
+           -alias <alias> -validity <validity>\
+           -dname "<distinguished-name>"\
+           -ext SAN=dns:<hostname1>,dns:<hostname2>,ip:<ipaddress1>
+   ```
+   
+   Example:
+   ```bash
+   $ keytool -storetype JKS -keystore server_unsigned.keystore.jks -storepass changeit\
+              -genkey -keyalg RSA -keysize 2048 -keypass changeit\
+              -alias server -validity 365\
+              -dname "CN=server.pravega.io, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown"\
```

Comment:
Consider examples that make clear what `C`, `ST`, `L`, `O` & `OU` mean, instead of using "Unknown" throughout, such as in the 1st `openssl` at the top of the document.

Reply:
Sure, I've made the change now.